### PR TITLE
Fix documentation about the SPLIT function

### LINKED ISSUE #1035: Documentation for SPLIT function is wrong
### Description

It should say:
```
SPLIT(Text, Index) - divides the provided text using space character as a separator and returns the substring at zero-based position specified by the second argument.
E.g.:
- SPLIT("Lorem ipsum dolor", 0) -> "Lorem"
- SPLIT("Lorem ipsum dolor", 1) -> "ipsum"
- SPLIT("Lorem ipsum dolor", 2) -> "dolor"
- SPLIT("Lorem ipsum dolor", 3) -> #VALUE
```

Also, add it to the known-limitations page as it is different from Google Sheets implementation. Consider renaming it.

### Links

- https://github.com/handsontable/hyperformula/issues/857
- https://handsoncode.slack.com/archives/C031GBX62SE/p1658825517285939 (internal discussion)

### Type of issue

- [ ] Missing documentation
- [x] Error in the documentation
- [ ] Request for a demo
- [ ] Request for a tutorial, guide etc.
- [ ] Other issues, suggestions or ideas

### PR DESCRIPTION
**Pull request by sequba**
### Context

- Update description in the built-in-functions guide
- Mention it in the known-limitations guide

### How has this been tested?

Local docs build

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Additional language file or change to the existing one (translations)
- [x] Change to the documentation

### Related issues:
1. Fixes #1035 
2.
3.

### Checklist:

- [x] My code follows the code style of this project.
- [ ] I described the modification in the CHANGELOG.md file.
- [ ] My change requires a change to the documentation.
- [ ] My change requires a migration guide.

### CODE REVIEW

File: docs/guide/known-limitations.md [hunk 1]
```diff
@@ -55,7 +55,8 @@ The inconsistencies are due to:
...
+| SPLIT function                                     | =SPLIT("Lorem ipsum dolor", 0)                                            | This function works differently from Google Sheets version but should be sufficient to achieve the same functionality in most scenarios. Read SPLIT function description in [the Built-in Functions page](built-in-functions.html#text).                                                               | Different syntax and return value.                                                                                                | No such function.                                        |
```
Comment (warpech) [inline docs/guide/known-limitations.md:59]:
```suggestion
| SPLIT function                                     | =SPLIT("Lorem ipsum dolor", 0)                                            | This function works differently from Google Sheets version but should be sufficient to achieve the same functionality in most scenarios. Read SPLIT function description on [the Built-in Functions page](built-in-functions.html#text).                                                               | Different syntax and return value.                                                                                                | No such function.                                        |
```

File: docs/guide/built-in-functions.md [hunk 2]
```diff
@@ -486,28 +486,28 @@ Total number of functions: **{{ $page.functionsCount }}**
...
+| RIGHT         | Extracts a given number of characters from the right side of a text string.                                                                                                                                                               | RIGHT("Text"; Number)                              |
+| SEARCH        | Returns the location of one text string inside another. (Allows the use of wildcards.)                                                                                                                                                    | SEARCH( "Text1"; "Text2"[; Number])                |
+| SPLIT         | Divides the provided text using space character as a separator and returns the substring at zero-based position specified by the second argument.<br>```SPLIT("Lorem ipsum", 0) -> "Lorem"```<br>```SPLIT("Lorem ipsum", 1) -> "ipsum"``` | SPLIT(Text; Index)                                 |
```
Comment (warpech) [inline docs/guide/built-in-functions.md:506]:
```suggestion
| SPLIT         | Divides the provided text using the space character as a separator and returns the substring at the zero-based position specified by the second argument.<br>```SPLIT("Lorem ipsum", 0) -> "Lorem"```<br>```SPLIT("Lorem ipsum", 1) -> "ipsum"``` | SPLIT(Text; Index)                                 |
```

**Review by warpech — CHANGES_REQUESTED** [review]
LGTM, just some grammar suggestions

File: docs/guide/known-limitations.md (hunk 1 above)
Comment (sequba) [reply docs/guide/known-limitations.md]: suggestion applied

File: docs/guide/built-in-functions.md (hunk 2 above)
Comment (sequba) [reply docs/guide/built-in-functions.md]: suggestion applied

**Review by warpech — APPROVED** [review]

**Pull request merged**